lua/wikis/commons/Infobox/Person.lua

```diff
@@ -451,6 +547,60 @@ function Person:displayLocations()
 	end)
 end
 
+---@param roleKey string
+---@return PersonRoleData?
+function Person:createRoleData(roleKey)
+	if String.isEmpty(roleKey) then return nil end
+
+	local key = roleKey:lower()
+	local roleData = ROLES[key]
+
+	if not roleData and key ~= '' then
+		local display = String.upperCaseFirst(roleKey)
+		roleData = {
+			display = display,
+			category = display .. "s"
+		}
+	end
+
+	return roleData
+end
+
+
+---@param roleData PersonRoleData?
+---@return string?
+function Person:_displayRole(roleData)
+	if not roleData then return end
+
+	---@param postFix string|integer|nil
+	---@return string?
+	local toDisplay = function(postFix)
+		postFix = postFix or ''
+		if not roleData['category' .. postFix] then return end
+		return Page.makeInternalLink(roleData['display' .. postFix], ':Category:' .. roleData['category' .. postFix])
+	end
+
+	local role1Display = toDisplay()
+	local role2Display = toDisplay(2)
+	local role3Display = toDisplay(3)
+
+	if role1Display then
+		if role2Display then
+			role2Display = '(' .. role2Display .. ')'
+			if role3Display then
+				role3Display = '(' .. role3Display .. ')'
+			end
+		end
+	end
```

Comment:
what if role1Display is set, role2Display is not set and role3Display is set?
it wouldn't add the brackets in that case